src/Identity/Core/src/SignInManager.cs

```diff
@@ -438,9 +473,13 @@ private async Task DoTwoFactorSignInAsync(TUser user, TwoFactorAuthenticationInf
             // When token is verified correctly, clear the access failed count used for lockout
             await ResetLockout(user);
 
+            var claims = new List<Claim>();
+            claims.Add(new Claim("amr", "mfa"));
```

Comment:
@brockallen The spec says if we have mfa we can also have specific authentication methods used, so would be expect to also see two more mfa claims of pwd, and otp here?

Reply:
It would require some more plumbing to flow things through, plus its user token provider specific what that is, its not always authenticator for example, that just happens to be what our token provider uses.  Unless we want to add that to the interface, its going to be a bit difficult

Reply:
Sure, you can have any number that mean whatever they need to. IdentityServer will also accept any number of `amr` claims, and in addition will interpret the ClaimsIdentity's AuthenticationType property as an `amr` as well.